scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -644,79 +580,66 @@ public Double doInTransaction(Session session) {
-                    return null;
-                }
-
-                Query query = session.getNamedQuery("getTotalNumberOfHostsUsed").setParameter("id", id);
-
-                return ((Long) query.uniqueResult()).intValue();
+        int result = executeReadOnlyTransaction(session -> {
+            Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
+            if (jobQuery.uniqueResult() == null) {
+                return null;
             }
 
+            Query query = session.getNamedQuery("getTotalNumberOfHostsUsed").setParameter("id", id);
+
+            return ((Long) query.uniqueResult()).intValue();
         });
 
         return checkResult(id, result);
     }
 
     public SchedulerAccount readAccount(final String username) {
-        return executeReadOnlyTransaction(new SessionWork<SchedulerAccount>() {
-
-            @Override
-            public SchedulerAccount doInTransaction(Session session) {
-                Query tasksQuery = session.getNamedQuery("readAccountTasks").setParameter("username", username);
+        return executeReadOnlyTransaction(session -> {
+            Query tasksQuery = session.getNamedQuery("readAccountTasks").setParameter("username", username);
 
-                int taskCount;
-                long taskDuration;
-
-                Object[] taskResult = (Object[]) tasksQuery.uniqueResult();
-                taskCount = ((Number) taskResult[0]).intValue();
-                if (taskResult[1] != null) {
-                    taskDuration = ((Number) taskResult[1]).longValue();
-                } else {
-                    taskDuration = 0;
-                }
+            int taskCount;
+            long taskDuration;
 
-                int jobCount;
-                long jobDuration;
+            Object[] taskResult = (Object[]) tasksQuery.uniqueResult();
+            taskCount = ((Number) taskResult[0]).intValue();
+            if (taskResult[1] != null) {
+                taskDuration = ((Number) taskResult[1]).longValue();
+            } else {
+                taskDuration = 0;
+            }
 
-                Query jobQuery = session.getNamedQuery("readAccountJobs").setParameter("username", username);
+            int jobCount;
+            long jobDuration;
 
-                Object[] jobResult = (Object[]) jobQuery.uniqueResult();
-                jobCount = ((Number) jobResult[0]).intValue();
-                if (jobResult[1] != null) {
-                    jobDuration = ((Number) jobResult[1]).longValue();
-                } else {
-                    jobDuration = 0;
-                }
+            Query jobQuery = session.getNamedQuery("readAccountJobs").setParameter("username", username);
 
-                return new SchedulerAccount(username, taskCount, taskDuration, jobCount, jobDuration);
+            Object[] jobResult = (Object[]) jobQuery.uniqueResult();
+            jobCount = ((Number) jobResult[0]).intValue();
```

Comment:
Why just one null check. Maybe you can find a way to write this as an explicit contract. Like you throw an InvalidYYYZZZException when the jobResult array has size 0